debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -381,11 +385,16 @@ protected void generateDeleteRecord(TableId tableId, Object[] oldRowData, Consum
         assert table != null;
 
         // check if we need to refresh our local schema due to DB schema changes for this table
-        if (refreshSchemaIfChanged && schemaChanged(messageList, table)) {
+        if (refreshSchemaIfChanged && schemaChanged(messageList, table, metadataInMessage)) {
             try (final PostgresConnection connection = taskContext.createConnection()) {
+                // Refresh the schema so we get information about primary keys
```

Comment:
You told me before, but what again is this information exactly about?

Reply:
The replication message itself does not bear infromation about what colums constitute the primary key - so we are getting it from JDBC - see https://github.com/eulerto/wal2json/issues/43

Reply:
Ok. So it seems useful to cut down the info we obtain via JDBC to the mininum and get everything else from the message. I understand it gets overridden later on, but the less moving pieces, the better :)

Reply:
I think it does not do any harm. And is it worthy to modify the core code to this functionality if this only interim solution till the PK info arrives to messages?